### PR TITLE
Fixes #26863 - adds task cancel

### PR DESCRIPTION
**Pull request by jjeffers**
This PR adds task cancellation for pulp 3 tasks.

To test this PR, you will need a Pulp v3 enabled smart proxy and pulp v3 server.

- Create a file repository with a URL of `https://repos.fedorapeople.org/repos/pulp/pulp/fixtures/file-many/`.
- Start a repository sync, then navigate to `/foreman_tasks/tasks`
- Locate the most recent "Synchronize repository..." task with a "pending" status, and open that task details page.
- You should observer an enabled button labelled "Cancel", and you should be able to click that button.
- The task should cancel and not complete the Sync operation with the pulp server.

### CODE REVIEW

**Comment by jlsherrill** [comment]
test failures due to https://github.com/Katello/katello/pull/8126

**Comment by jjeffers** [comment]
[test katello]

File: app/lib/actions/pulp3/abstract_async_task.rb [hunk 1]
```diff
@@ -82,7 +82,14 @@ def cancel!
       end
 
       def cancel
-        fail "Cancelling not supported"
+        output[:pulp_tasks].each do |pulp_task|
+          data = PulpcoreClient::Task.new
+          pulp3_api.tasks_api.tasks_cancel(pulp_task['_href'], data)
```
Comment (jlsherrill) [inline app/lib/actions/pulp3/abstract_async_task.rb]: why did you have to pass in a 'new' task object?  was it related to https://pulp.plan.io/issues/4883 ?

**Comment by jjeffers** [comment]
This needs to be tested with the updated pulp file client at version https://rubygems.org/gems/pulp_file_client/versions/0.1.0b1.dev.1560187476

File: katello.gemspec [hunk 2]
```diff
@@ -42,7 +42,7 @@ Gem::Specification.new do |gem|
   # Pulp
   gem.add_dependency "runcible", ">= 2.11.0", "< 3.0.0"
   gem.add_dependency "anemone"
-  gem.add_dependency "pulpcore_client"
+  gem.add_dependency "pulpcore_client", "3.0.0rc2.dev.1560272824"
```
Comment (johnpmitsch) [inline katello.gemspec]: Is this an intended change in the PR?

Comment (jjeffers) [reply katello.gemspec]: Yes, this is due to how the pulpcore client versions were specified by the pulp3 team. The labels are considered to be "earlier" than than a previous version (May 20, 2019 release date) labelled as "3.0.0rc2", despite a June 11 2019 release date. Until the pulp3 removes the mislabelled gem versions, we have to pin the pulpcore client.

**Comment by jlsherrill** [comment]
This gem pin will break nightly rpm builds, maybe less than 3.0.0.rc2?  I'm
not quite sure

**Comment by ianballou** [comment]
It does still work with `gem.add_dependency "pulpcore_client", "< 3.0.0rc2"`

**Comment by johnpmitsch** [comment]
I'm told the versioning issue is sorted out so merging!

**Pull request merged**